src/transformers/modeling_utils.py

```diff
@@ -2063,12 +2075,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
             logger.info("Detected DeepSpeed ZeRO-3: activating zero.init() for this model")
             init_contexts = [deepspeed.zero.Init(config_dict_or_path=deepspeed_config())] + init_contexts
+        elif load_in_8bit:
+            init_contexts = [init_empty_weights_8bit()]  # Force enable init empty weights
+            logger.info("Detected 8-bit loading: activating 8-bit loading for this model")
```

Comment:
I don't understand why we need a different function here, as the linear layers have not been replaced yet.

Reply:
Actually my bad, these functions are needed. As far as I understood from my experiments a bug occurs there because `init_empty_weights` force assign the weights to be `nn.Parameter` when recursing on all modules, including the `Linear8bitLt` layers. So at the moment we put the modules on the meta device, we have to make sure their parameters are from the class `bnb.nn.Int8Params` I think

Reply:
So we do not need it here specifically but will need it inside the `replace_8bit` function !

Reply:
At which point do we have `Linear8bitLt` layers since the conversion happens after the initialization?

Reply:
So for now it happens here: https://github.com/younesbelkada/transformers/blob/b92c25c3835f37667ee033a2b406846da06f16f4/src/transformers/utils/bitsandbytes.py#L91

Reply:
And planning to use the `init_empty_weights_8bit` context manager there

Reply:
@sgugger there might be a simpler option: by replacing [theses](https://github.com/huggingface/accelerate/blob/70ca65a9a1c336fe02800a0bb319ccbf4b9c6d26/src/accelerate/big_modeling.py#L68) lines by:
```
param_cls = type(module._parameters[name])
kwargs = module._parameters[name].__dict__
module._parameters[name] = param_cls(module._parameters[name].to(torch.device(“meta”)), **kwargs)
```
everything works smoothly, wdyt about opening a PR there?

Reply:
Works for me as long as it works for the special subclasses used by bitsandbytes!